iceoryx_hoofs/include/iceoryx_hoofs/cxx/variant.hpp

```diff
@@ -259,8 +261,11 @@ class variant
     constexpr uint64_t index() const noexcept;
 
   private:
-    alignas(algorithm::max(alignof(Types)...)) internal::byte_t m_storage[TYPE_SIZE]{0u};
-    uint64_t m_type_index = INVALID_VARIANT_INDEX;
+    /// @todo #1196 Replace with UninitializedArray
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays)
+    alignas(algorithm::max(alignof(Types)...)) internal::byte_t m_storage[TYPE_SIZE]{0U};
```

Comment:
We could also use `std::aligned_storage` here to get rid of the warning.

Reply:
@elBoberido I think most compiler will implement `aligned_storage` with an array, see [cppref's implementation example](https://en.cppreference.com/w/cpp/types/aligned_storage).